design/round1/04-auth-authorization-detailed.md

```diff
@@ -0,0 +1,180 @@
+- Enforce tenant isolation by filtering queries on `tenant_id`.
+
+## Authorization: RBAC Policy
+
+### Roles
+- owner: full control; manage members; delete board; manage generations.
+- editor: create/update generations; edit board metadata; invite/remove members of lower role if allowed.
+- viewer: read-only access to board and generations.
+
+### Permission Matrix (selected)
+- Board
+  - create: authenticated user → owner of new board
+  - read: owner/editor/viewer OR `boards.is_public = true`
+  - update: owner or editor
+  - delete: owner
+  - members: owner or editor (read); manage: owner
+- Generation
+  - create: editor or owner on target board
+  - read: viewer/editor/owner, or public board
+  - cancel/delete: owner or editor if created by self; owner can manage all
+
+### Enforcement Points
+- GraphQL resolvers and mutations must call helpers:
+```python
+async def require_board_role(db, board_id: UUID, user_id: UUID, roles: set[str]): ...
+async def can_read_board(db, board_id: UUID, user_id: UUID | None) -> bool: ...
+```
+- Storage presigned URLs only issued if `can_read_board`/`require_board_role` passes.
+- SSE streams for job progress validate `can_read_board`.
+
+## Backend Integration in Strawberry/FastAPI
+
+### Context Injection
+```python
+# graphql/schema.py
+from strawberry.fastapi import GraphQLRouter
+
+async def get_context_dependency(request: Request) -> dict:
+    return {
+        "auth": request.state.auth_context,  # AuthContext
+        "db": request.state.db,
+    }
+
+router = GraphQLRouter(schema, context_getter=get_context_dependency)
+```
+
+### Resolver Usage
+```python
+@strawberry.type
+class Query:
+    @strawberry.field
+    async def board(self, info, id: UUID) -> Optional[Board]:
+        auth = info.context["auth"]
+        if not await can_read_board(info.context["db"], id, auth.user.id if auth.user else None):
+            raise PermissionError("Not authorized")
+        return await boards_repo.get_by_id(id)
+```
+
+### Mutations
+- `create_board`: authenticated required; set `owner_id = auth.user.id` and `tenant_id`.
+- `add_board_member`, `update_board_member_role`, `remove_board_member`: require owner.
+- `create_generation`: require `editor` or `owner` on `board_id`.
+
+## Frontend: `useAuth()` and Providers
+
+### Stable Hook Contract
+```ts
+interface AuthState {
+  user: {
+    id: string
+    email?: string
+    displayName?: string
+    avatarUrl?: string
+    provider: string
+    subject: string
+  } | null
+  status: 'unauthenticated' | 'loading' | 'authenticated'
+  signIn: (opts?: Record<string, unknown>) => Promise<void>
+  signOut: () => Promise<void>
+  getToken: () => Promise<string | null>
+}
+```
+
+- Implementations:
+  - `@toolkit/auth-supabase`: wraps Supabase JS; maps session user → above shape.
+  - `@toolkit/auth-clerk`: wraps Clerk JS.
+  - `@toolkit/auth-auth0`: wraps Auth0 SPA SDK.
+  - `@toolkit/auth-jwt`: local token storage with backend-issued JWT.
```

Comment:
I think that these should start with `@weirdfingers` rather than `@toolkit` as per packages/frontend/package.json